zcash_client_backend/src/fees/common.rs

```diff
@@ -530,13 +519,13 @@ where
                         // Zero-valued change is also always allowed for this policy, but when
                         // no change memo is given, we might omit the change output instead.
 
-                        let fee_with_dust = (total_in - total_out)
+                        let fee_with_dust = (total_in - subtotal_out)
                             .expect("we already checked for sufficient funds");
```

Comment:
`subtotal_out <= total_out`, so this is correct once we've added back the check at lines 460-463. However, since `total_change == total_in - (subtotal_out + total_fee)`, we can also calculate it like this:
```suggestion
                        let fee_with_dust = (total_change + total_fee).ok_or_else(overflow)?
```
This is better because it makes the assertion redundant.